ocaml/xcp-rrdd/bin/rrdd/rrdd_server.ml

```diff
@@ -491,6 +491,33 @@ let dump_host_dss_to_file (file : string) : unit =
       flush oc
   )
 
+let dump_vm_dss_to_file (file : string) : unit =
```

Comment:
This needs to be protected by the mutex?

Reply:
Could you elaborate what needs a mutex?

Reply:
Look at it again, I think it is only `vm_rrds`.

Reply:
Thanks for remainder @minglumlu, I have refactored the code and add the `mutex` for every `vm_rrds`